### PR TITLE
Provide profile to pipeline

### PR DESCRIPTION
**Pull request by liuyehcf**
Pipeline support profile

### CODE REVIEW

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 1]
```diff
@@ -184,16 +187,34 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
                 } else {
                     scan_operator->set_io_threads(nullptr);
                 }
+                for (int32_t j = operators.size() - 1; j >= 0; --j) {
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/fragment_executor.cpp:190]: Using `operators` after moved in L182 is undefined behavior. Maybe we could move L190~L198 before L179.

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 2]
```diff
@@ -184,16 +187,34 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
...
         } else {
             if (is_root) {
                 _fragment_ctx->set_num_root_drivers(degree_of_parallelism);
             }
-            for (auto i = 0; i < degree_of_parallelism; ++i) {
+            for (size_t i = 0; i < degree_of_parallelism; ++i) {
                 auto&& operators = pipeline->create_operators(degree_of_parallelism, i);
                 DriverPtr driver =
                         std::make_shared<PipelineDriver>(std::move(operators), _query_ctx, _fragment_ctx, i, is_root);
+                for (int32_t j = operators.size() - 1; j >= 0; --j) {
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/fragment_executor.cpp:209]: Similar as above.

File: be/src/exec/pipeline/pipeline_driver.cpp [hunk 3]
```diff
@@ -192,7 +192,7 @@ void PipelineDriver::finalize(RuntimeState* runtime_state, DriverState state) {
     }
 }
 
-std::string PipelineDriver::to_debug_string() const {
+std::string PipelineDriver::to_debug_string() {
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/pipeline_driver.cpp:195]: Why removing `const`?

File: be/src/exec/pipeline/pipeline.h [hunk 4]
```diff
@@ -50,6 +50,20 @@ class Pipeline {
         }
     }
 
+    std::string to_debug_string() {
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/pipeline.h:53]: Maybe we can mark `to_debug_string` as `const`?

File: be/src/exec/pipeline/olap_chunk_source.cpp [hunk 5]
```diff
@@ -257,10 +297,70 @@ Status OlapChunkSource::_read_chunk_from_storage(RuntimeState* state, vectorized
...
+    COUNTER_UPDATE(_index_load_timer, _reader->stats().index_load_ns);
+
+    COUNTER_UPDATE(_total_pages_num_counter, _reader->stats().total_pages_num);
+    COUNTER_UPDATE(_cached_pages_num_counter, _reader->stats().cached_pages_num);
+
+    COUNTER_UPDATE(_bi_filtered_counter, _reader->stats().rows_bitmap_index_filtered);
+    COUNTER_UPDATE(_bi_filter_timer, _reader->stats().bitmap_index_filter_timer);
+    COUNTER_UPDATE(_block_seek_counter, _reader->stats().block_seek_num);
+
+    // COUNTER_SET(_pushdown_predicates_counter, (int64_t)_params.predicates.size());
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply be/src/exec/pipeline/olap_chunk_source.cpp]: fixed. comment removed

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 6]
```diff
@@ -160,6 +172,9 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
         const auto& pipeline = pipelines[n];
         // DOP(degree of parallelism) of Pipeline's SourceOperator determines the Pipeline's DOP.
         const auto degree_of_parallelism = pipeline->source_operator_factory()->degree_of_parallelism();
+        LOG(INFO) << "Pipeline " << pipeline->to_debug_string() << " parallel=" << degree_of_parallelism
```
Comment (satanson) [inline be/src/exec/pipeline/fragment_executor.cpp:175]: use VLOG_ROW instead

Comment (kangkaisen) [inline be/src/exec/pipeline/fragment_executor.cpp:175]: This log seems repeated with    

```
LOG(INFO) << "Prepare(): query_id=" << print_id(query_id)
              << " fragment_instance_id=" << print_id(params.fragment_instance_id) << " backend_num=" << backend_num;
```

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Pull request merged**